### ISSUE TITLE
Core dump when opening "Settings" tab in "Preferences" when global "gui.editor" is not configured

### ISSUE THREAD

**Issue by cherio** [git-cola/git-cola#1263]
With the new version I can not longer access `Settings` tab in the `File` -> `Preferences` dialog. Every time I try to switch to that tab the application crashes producing core dump and the following message:

```
Traceback (most recent call last):
  File "/usr/lib/python3.10/site-packages/cola/widgets/prefs.py", line 412, in update_widget
    widget.update_from_config()
  File "/usr/lib/python3.10/site-packages/cola/widgets/prefs.py", line 276, in update_from_config
    FormWidget.update_from_config(self)
  File "/usr/lib/python3.10/site-packages/cola/widgets/prefs.py", line 103, in update_from_config
    set_widget_value(widget, value)
  File "/usr/lib/python3.10/site-packages/cola/widgets/prefs.py", line 112, in set_widget_value
    widget.setText(value)
TypeError: setText(self, str): argument 1 has unexpected type 'function'
Aborted (core dumped)
```
This is happening with the [latest Arch/AUR](https://aur.archlinux.org/packages/git-cola) install.

I am not sure if this is related but coincidentally this same instance/install produces errors (without completely crashing) whenever I try to `Launch Editor` from either the regular file status list (also a `TypeError`) ...

```
Traceback (most recent call last):

  File "/usr/lib/python3.10/site-packages/cola/cmds.py", line 2965, in do
    return cmd.do()

  File "/usr/lib/python3.10/site-packages/cola/cmds.py", line 1413, in do
    if fnmatch(editor, pattern):

  File "/usr/lib/python3.10/fnmatch.py", line 40, in fnmatch
    name = os.path.normcase(name)

  File "/usr/lib/python3.10/posixpath.py", line 54, in normcase
    return os.fspath(s)

TypeError: expected str, bytes or os.PathLike object, not function
```
... or from `DAG` ...
```
Traceback (most recent call last):

  File "/usr/lib/python3.10/site-packages/cola/cmds.py", line 2965, in do
    return cmd.do()

  File "/usr/lib/python3.10/site-packages/cola/cmds.py", line 1418, in do
    core.fork(utils.shell_split(editor) + opts)

  File "/usr/lib/python3.10/site-packages/cola/utils.py", line 284, in shell_split
    values = _shell_split_py3(s)

  File "/usr/lib/python3.10/site-packages/cola/utils.py", line 271, in _shell_split_py3
    result = shlex.split(s)

  File "/usr/lib/python3.10/shlex.py", line 315, in split
    return list(lex)

  File "/usr/lib/python3.10/shlex.py", line 300, in __next__
    token = self.get_token()

  File "/usr/lib/python3.10/shlex.py", line 109, in get_token
    raw = self.read_token()

  File "/usr/lib/python3.10/shlex.py", line 140, in read_token
    nextchar = self.instream.read(1)

AttributeError: 'function' object has no attribute 'read'
```

**Comment by cherio** [comment]
I figured out the `Launch Editor` error. There might be an issue on line 1393 in `/usr/lib/python3.10/site-packages/cola/cmds.py`:

```
        if self.background_editor:
            editor = prefs.background_editor(context)
        else:
            editor = prefs.editor(context)
```
With pure guess I made the `Launch Editor` error go away by adding `gui.editor=/path/to/editor` into the repository `.git/config`. IMO I shouldn't have to do that, it should work without that attribute.

I can't figure out the cause of the core dump issue though. I am not a python developer. It also looks like a failure to correctly initialize from configuration (93 in `/usr/lib/python3.10/site-packages/cola/widgets/prefs.py`) but I am out of my depth here:

```
    def update_from_config(self):
        if self.source == 'user':
            getter = self.cfg.get_user_or_system
        else:
            getter = self.cfg.get

        for config, widget in self.widget_to_config.items():
            value = getter(config)
            if value is None:
                value = self.defaults[config]
            set_widget_value(widget, value)
```

**Comment by cherio** [comment]
There is definitely something wrong with reading configuration and the defaults, specifically editor. I was able to make core dump error go away by adding the `gui.editor` into `$HOME/.gitconfig`:
```
[gui]
        editor = /path/to/editor
```
It looks like `self.defaults[config]` returns a `function` for `gui.editor` on line 102 in `/usr/lib/python3.10/site-packages/cola/widgets/prefs.py` when editor parameter is not explicitly configured.

As I mentioned before I shouldn't have to have that configuration parameter in either global or repository config in order for `git-cola` to work and not throw errors. This looks like a bug to me.

**Comment by davvid** [comment]
Thanks for digging in. Indeed, that change was related. Thanks for the heads-up ~ I was able to reproduce the error and it's fixed now. This fix will be in the v4.0.3+ release. Please test the latest from git.
